azurecosmos/src/main/java/site/ycsb/db/AzureCosmosClient.java

```diff
@@ -40,120 +62,156 @@
-            DEFAULT_CONNECTION_MODE);
-
-    ConnectionPolicy connectionPolicy = new ConnectionPolicy();
-    connectionPolicy.setEnableEndpointDiscovery(false);
-    connectionPolicy.setConnectionMode(ConnectionMode.valueOf(connectionModeString));
-    connectionPolicy.setMaxPoolSize(this.getIntProperty("azurecosmos.maxConnectionPoolSize",
-                                                        connectionPolicy.getMaxPoolSize()));
-    connectionPolicy.setIdleConnectionTimeout(this.getIntProperty("azurecosmos.idleConnectionTimeout",
-                                                                  connectionPolicy.getIdleConnectionTimeout()));
-
-    RetryOptions retryOptions = new RetryOptions();
+    AzureCosmosClient.userAgent = this.getStringProperty("azurecosmos.userAgent", DEFAULT_USER_AGENT);
+
+    AzureCosmosClient.useUpsert = this.getBooleanProperty("azurecosmos.useUpsert", DEFAULT_USE_UPSERT);
+
+    AzureCosmosClient.databaseName = this.getStringProperty("azurecosmos.databaseName", DEFAULT_DATABASE_NAME);
+
+    AzureCosmosClient.maxDegreeOfParallelism = this.getIntProperty("azurecosmos.maxDegreeOfParallelism",
+        DEFAULT_MAX_DEGREE_OF_PARALLELISM);
+
+    AzureCosmosClient.maxBufferedItemCount = this.getIntProperty("azurecosmos.maxBufferedItemCount",
+        DEFAULT_MAX_BUFFERED_ITEM_COUNT);
+
+    AzureCosmosClient.preferredPageSize = this.getIntProperty("azurecosmos.preferredPageSize",
+        DEFAULT_PREFERRED_PAGE_SIZE);
+
+    AzureCosmosClient.includeExceptionStackInLog = this.getBooleanProperty("azurecosmos.includeExceptionStackInLog",
+        DEFAULT_INCLUDE_EXCEPTION_STACK_IN_LOG);
+
+    ConsistencyLevel consistencyLevel = ConsistencyLevel.valueOf(
+        this.getStringProperty("azurecosmos.consistencyLevel", DEFAULT_CONSISTENCY_LEVEL.toString().toUpperCase()));
+    boolean useGateway = this.getBooleanProperty("azurecosmos.useGateway", DEFAULT_USE_GATEWAY);
+
+    ThrottlingRetryOptions retryOptions = new ThrottlingRetryOptions();
     retryOptions.setMaxRetryAttemptsOnThrottledRequests(this.getIntProperty(
-            "azurecosmos.maxRetryAttemptsOnThrottledRequests",
-            retryOptions.getMaxRetryAttemptsOnThrottledRequests()));
-    retryOptions.setMaxRetryWaitTimeInSeconds(this.getIntProperty(
-              "azurecosmos.maxRetryWaitTimeInSeconds",
-              retryOptions.getMaxRetryWaitTimeInSeconds()));
-    connectionPolicy.setRetryOptions(retryOptions);
+        "azurecosmos.maxRetryAttemptsOnThrottledRequests", retryOptions.getMaxRetryAttemptsOnThrottledRequests()));
+    retryOptions.setMaxRetryWaitTime(Duration.ofSeconds(this.getIntProperty("azurecosmos.maxRetryWaitTimeInSeconds",
+        Math.toIntExact(retryOptions.getMaxRetryWaitTime().toMillis() / 1000))));
+
+    DirectConnectionConfig directConnectionConfig = new DirectConnectionConfig();
+    directConnectionConfig.setMaxConnectionsPerEndpoint(this.getIntProperty("azurecosmos.maxConnectionPoolSize",
```

Comment:
let's not reuse the same config name for both GW and direct mode, this is very different from maxConnectionPoolSize

config name should be `azurecosmos.directMaxConnectionsPerEndpoint`

also if the config doesn't exist we should not set the config from the sdk.